components/RoomEditForm/index.js

```diff
@@ -0,0 +1,80 @@
+import { useState } from "react";
+import Heading from "../Heading";
+import { FormContainer, Label, Input, Form, SubmitButton } from "./styles";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+export default function RoomEditForm({ room }) {
+  const router = useRouter();
+  const [formData, setFormData] = useState({});
+
+  useEffect(() => {
+    setFormData({
+      roomName: room?.name || "",
+      roomSubject: room?.subject || "",
+    });
+  }, [room]);
```

Comment:
This does work, but you can make this an even easier by setting the initial state of your `formData`.
Instead of initializing it as an empty object you can pass the room prop as argument.
```suggestion
export default function RoomEditForm({ room }) {
  const router = useRouter();
  const [formData, setFormData] = useState({
      roomName: room?.name || "",
      roomSubject: room?.subject || "",
    });

```

Something like this.

Reply:
Thats because I need the default values when opening the edit form. With your suggested code I do not see default values.

Reply:
Mh, I see. All right, if you can't make it work, leave it unchanged 👍🏼